x-pack/plugin/core/src/main/java/org/elasticsearch/license/ClusterStateLicenseService.java

```diff
@@ -242,10 +245,15 @@ protected PutLicenseResponse newResponse(boolean acknowledged) {
                 @Override
                 public ClusterState execute(ClusterState currentState) throws Exception {
                     XPackPlugin.checkReadyForXPackCustomMetadata(currentState);
-                    final Version oldestNodeVersion = currentState.nodes().getSmallestNonClientNodeVersion();
-                    if (licenseIsCompatible(newLicense, oldestNodeVersion) == false) {
+                    int maxCompatibleLicenseVersion = LicenseUtils.getMaxLicenseVersion();
+                    if (newLicense.version() <= maxCompatibleLicenseVersion == false) {
```

Comment:
Could this be inverted to `>`?

Reply:
I switched this to `>=` but it needs to include the equality.

Reply:
Hrm, but that isn't equivalent then. `(a <= b) == false` is the equivalent of `a > b`.

Reply:
Err thanks, whiffed on the logic - that's what I get for doing this late in the evening :p

You're right, I'll fix this before merging.